src/automation.js

```diff
@@ -0,0 +1,97 @@
+import { rdfs, rdf } from '@tpluscode/rdf-ns-builders'
+import $rdf from '@rdfjs/dataset'
+import clownface from 'clownface'
+import TermMap from '@rdfjs/term-map'
+import * as dimensionQueries from './queries/dimensions.js'
+import * as ns from './ns.js'
+
+export async function generateDimensions(view, queries = dimensionQueries) {
```

Comment:
This is the main new functionality, which modifies the view to (re-)generate automatic dimensions

Reply:
`queries` arg is optional to allow injecting in test to avoid running SPARQL in mocha